restore $attachment_ids for product thumbnails template

rrennick opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This is a followup PR to #27289. It updates the template versions to the correct version and fixes a warning created in the original PR. Ref https://github.com/woocommerce/woocommerce/commit/e49527cf9d546985a4f9c6db284d9540de302efb#r43055263

### How to test the changes in this Pull Request:

1. Create a product and assign multiple gallery images
2. View the single product
3. First gallery image should be used as product image and not be duplicated in the list of gallery images
4. Assign a product image
5. Product image & gallery should be displayed the same as in master
6. Image zoom should work in all testing
7. There should be no debug warnings logged.


### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

Patches a fix for 4.7

rrennick committed:
restore $attachment_ids for product thumbnails template

rrennick committed:
Merge branch 'master' into fix/template-attachment-ids

claudiosanches commented on includes/wc-template-functions.php:
```diff
@@ -1477,13 +1477,15 @@ function woocommerce_pagination() {
 		}
-		wc_get_template( 'single-product/product-thumbnails.php', array( 'post_thumbnail_id' => $post_thumbnail_id ) );
+		$args = compact( 'post_thumbnail_id', 'gallery_image_ids' );
+		wc_get_template( 'single-product/product-thumbnails.php', $args );
```
```suggestion
		wc_get_template( 'single-product/product-image.php', $args );
```

claudiosanches committed:
Update includes/wc-template-functions.php

claudiosanches committed:
Fixed PHP notice

> PHP Notice: compact(): Undefined variable: gallery_image_ids in includes/wc-template-functions.php on line 1487

claudiosanches reviewed (approved):
Everything fixed and working as expected now.
Thanks!
